token/client/src/token.rs

```diff
@@ -210,12 +221,29 @@ where
         &self.pubkey
     }
 
-    pub fn with_payer<S2: Signer>(&self, payer: S2) -> Token<T, S2> {
+    pub fn with_payer(&self, payer: Arc<dyn Signer>) -> Token<T> {
         Token {
             client: Arc::clone(&self.client),
             pubkey: self.pubkey,
             payer,
             program_id: self.program_id,
+            nonce_account: self.nonce_account,
+            nonce_authority: self.nonce_authority,
+            // XXX FIXME this is wrong but i dont understand the implications of clone on RwLock
+            // in fact i dont understand why we use write at all why not just Option<String> and clone
+            memo: Arc::new(RwLock::new(None)),
+        }
+    }
+
+    pub fn with_nonce(&self, nonce_account: &Pubkey, nonce_authority: &Pubkey) -> Token<T> {
```

Comment:
Careful here, there's an inconsistency in the interface between `with_payer` and `with_memo`, where `with_payer` creates a new `Token`, but `with_memo` modifies the existing one.  Because of that, this is incorrectly used in the CLI.

At first, I wrote that this should work like `with_memo` and modify the underlying client, but after reading this over, I think we're better off making this a factory-style constructor as you have it, so you can do `let token = with_payer().with_nonce()`.

Reply:
yes youre right i use `with_nonce` (my own function!!) wrong h no